packages/docker/src/utils.ts

```diff
@@ -9,7 +9,7 @@ const exec = require('@actions/exec')
 export interface RunDockerCommandOptions {
   workingDir?: string
   input?: Buffer
-  env?: { [key: string]: string }
+  env?: { [key: string]: string | undefined }
```

Comment:
Oh sorry I suggested wrong
Can we not change this type to match the toolkits exec deffinition: https://github.com/actions/toolkit/blob/d2b7d85e7cb61467395604aae1b84439c320cb84/packages/exec/src/interfaces.ts#L10

May I suggest just doing below: 
```js
if (process.env.PATH) {
    newOptions.env["PATH"] = process.env.PATH
}
```

Reply:
I will make this change, but it should be noted that that actions/toolkit type is not correct. It states `env` defaults to `process.env`, but `nodejs.ProcessEnv` type from `@types/node` (and the `process.env` object itself) does, in fact allow for `undefined`.

Could even describe `env` as `typeof env` (which will then infer the type from process directly) and see that `undefined` is still allowed

Reply:
I agree with you :relaxed: 
But just for the sake of consistency between packages, I would rather do it this way and change it later in both places than have a mismatch